client/components/plans/plan-actions/index.jsx

```diff
@@ -142,6 +140,21 @@ module.exports = React.createClass( {
 		return true;
 	},
 
+	shouldOfferFreeTrial: function() {
+		if ( config.isEnabled( 'upgrades/free-trials' ) ) {
+			if ( this.props.isInSignup && this.props.flowName !== 'free-trial' ) {
```

Comment:
I think it's a little strange to check one prop only if another prop is true. `this.props` will always be an object in this case, so the `isInSignup` check isn't necessary here.